File: q3/q3.py
def isPrime(n):
    if n == 1 or n == 0:
        return False
    else:
        for i in range(1, n):
            if n % i == 0:
                return False
            else:
                return True

n = 600851475143
hpf = 0
for i in range(1, n):
    if isPrime(i):
        if n % i == 0:
            hpf = i
print(hpf)

File: q3/q3.c
//doesn't work??

#include <stdbool.h>
#include <stdio.h>
#include <math.h>
  
bool isPrime(int n)
{
    if (n == 1 || n == 0)
        return false;
    for (int i = 2; i * i <= n; i++) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

int main()
{
    int n = 600851475143;
    int hpf = 0;
    for (int i = 2; i < n; i++) {
        if (isPrime(i)) {
            if (n%i == 0) {
                hpf = i;
            }
        }
    }
    printf("%d\n", hpf);
}